### ISSUE TITLE
Can't launch 0.5.0, Win 10

### ISSUE THREAD

**Issue by slartie** [axcore/tartube#10]
Notes:
====
I used the 64bit installer
I'm using Windows 10
Both of the dll files mentioned in the error report are present in the folder.

Popup dialog error output:
==================
Titlebar: python3.exe - Entry Point Not Found
Content: The procedure entry point inflateValidate could not be located in the dynamic link library C:\Tartube\msys64\mingw64\bin\libpng16-16.dll

Console output:
===========
C:\Tartube\msys64\home\user\tartube>..\..\..\mingw64\bin\python3.exe tartube.py

** (python3.exe:9052): WARNING **: 23:01:32.700: Failed to load shared library 'libgdk-3-0.dll' referenced by the typelib: 'libgdk-3-0.dll': The specified procedure could not be found.
Traceback (most recent call last):
  File "tartube.py", line 32, in <module>
    from lib import mainapp
  File "C:\Tartube\msys64\home\user\tartube\lib\mainapp.py", line 26, in <module>
    from gi.repository import Gtk, GObject, GdkPixbuf
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\importer.py", line 145, in load_module
    importlib.import_module('gi.repository.' + dep.split("-")[0])
  File "C:\Tartube\msys64\mingw64\lib\python3.7\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\importer.py", line 146, in load_module
    dynamic_module = load_overrides(introspection_module)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\__init__.py", line 118, in load_overrides
    override_mod = importlib.import_module(override_package_name)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\Gdk.py", line 83, in <module>
    Color = override(Color)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\__init__.py", line 195, in override
    assert g_type != TYPE_NONE
AssertionError

C:\Tartube\msys64\home\user\tartube>PAUSE
Press any key to continue . . .

**Comment by axcore** [comment]
Can you run the Gtk graphics test? It's in the Tartube folder in the windows Start Menu.

**Comment by slartie** [comment]
Sure thing. This is the result.


C:\Tartube\msys64\home\user\tartube>..\..\..\mingw64\bin\python3.exe hello_world.py

** (python3.exe:7076): WARNING **: 15:11:15.536: Failed to load shared library 'libgdk-3-0.dll' referenced by the typelib: 'libgdk-3-0.dll': The specified procedure could not be found.
Traceback (most recent call last):
  File "hello_world.py", line 3, in <module>
    from gi.repository import Gtk
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\importer.py", line 145, in load_module
    importlib.import_module('gi.repository.' + dep.split("-")[0])
  File "C:\Tartube\msys64\mingw64\lib\python3.7\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\importer.py", line 146, in load_module
    dynamic_module = load_overrides(introspection_module)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\__init__.py", line 118, in load_overrides
    override_mod = importlib.import_module(override_package_name)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\Gdk.py", line 83, in <module>
    Color = override(Color)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\__init__.py", line 195, in override
    assert g_type != TYPE_NONE
AssertionError

C:\Tartube\msys64\home\user\tartube>PAUSE
Press any key to continue . . .

**Comment by axcore** [comment]
Hmmm.

The method I've been using to create the Windows installer is the one described in

https://bitbucket.org/sirdeniel/deploy-python-gtk3-apps-for-windows/

It looks like this isn't reliable enough, especially not for Windows 10 users, so perhaps I will have to find some completely different installation method.

**Comment by slartie** [comment]
If you need me to test something out, just let me know.

**Comment by axcore** [comment]
The installer for v0.6.0 has been redesigned. Please test it, and tell me if it works.

**Comment by slartie** [comment]
Unfortunately no luck. This time there was no popup error, but the console error is as follows:

C:\Tartube\msys64\home\user\tartube>..\..\..\mingw64\bin\python3.exe tartube

** (python3.exe:10464): WARNING **: 19:08:46.772: Failed to load shared library 'libgdk-3-0.dll' referenced by the typelib: 'libgdk-3-0.dll': The specified procedure could not be found.
Traceback (most recent call last):
  File "tartube", line 32, in <module>
    from lib import mainapp
  File "C:\Tartube\msys64\home\user\tartube\lib\mainapp.py", line 26, in <module>
    from gi.repository import Gtk, GObject, GdkPixbuf
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\importer.py", line 145, in load_module
    importlib.import_module('gi.repository.' + dep.split("-")[0])
  File "C:\Tartube\msys64\mingw64\lib\python3.7\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\importer.py", line 146, in load_module
    dynamic_module = load_overrides(introspection_module)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\__init__.py", line 118, in load_overrides
    override_mod = importlib.import_module(override_package_name)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\Gdk.py", line 83, in <module>
    Color = override(Color)
  File "C:\Tartube\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\__init__.py", line 195, in override
    assert g_type != TYPE_NONE
AssertionError

C:\Tartube\msys64\home\user\tartube>PAUSE
Press any key to continue . . .

**Comment by axcore** [comment]
Have you installed anything else on your system that uses Gtk?

**Comment by slartie** [comment]
I just performed a system wide search for the gtk libraries, just to be sure. Nothing there apart from the tartube installation.

**Comment by axcore** [comment]
Ok, so it's not a clash between different versions of Gtk. Perhaps you can try this:

Find the tartube_64bit.bat file. It should be in C:\Tartube\msys64\home\user\tartube\

Change the first line to this:

..\..\..\mingw64\bin\python3.7.exe tartube

Then try to start Tartube as normal

**Comment by slartie** [comment]
No luck. ".......\" doesn't work in Windows, so I tried with the absolute path as well. It still results in the libgdk error.

**Comment by axcore** [comment]
Oh dear, git swallowed by backslashes. The only change was python3.exe to python3.7.exe

Do you feel like trying a manual installation? It would take about 10-15 minutes (but most of that time is waiting, not typing). The steps are:

1. Install msys2 from www.msys2.org (it's a Linux emulator for Windows). You need the 64bit version, msys2-x86_64-2019nnnn.exe
2. Copy the Tartube source code into C:\msys2\home\YOURNAME\, so the main tartube script would then be C:\msys2\home\YOURNAME\tartube\tartube
3. Run the terminal window, C:\msys2\mingw64.exe
4. In the new terminal window, type: pacman -Syu
5. The terminal window will want to close down. Close it, and then restart it
6. Type these commands, one by one:
pacman -Su
pacman -S mingw-w64-x86_64-python3 
pacman -S mingw-w64-x86_64-python3-pip
pacman -S mingw-w64-x86_64-python3-gobject
pacman -S mingw-w64-x86_64-python3-requests
pacman -S mingw-w64-x86_64-gtk3 
pacman -S mingw-w64-x86_64-gsettings-desktop-schemas
cd tartube
python3 hello_world.py
python3 tartube

**Comment by slartie** [comment]
So far so good. It works from inside the msys environment, which I honestly suspected it would.

Progress.

**Comment by axcore** [comment]
Great! Now let's try it without the terminal window.

Go into the folder C:\msys2\home\YOURNAME\tartube\nsis

There are four windows .BAT files there. Move all of them into the folder above, i.e. C:\msys2\home\YOURNAME\tartube

Then, try running tartube_64bit.bat

**Comment by slartie** [comment]
We're back to this:

C:\msys64\home\slart\tartube>..\..\..\mingw64\bin\python3.exe tartube

** (python3.exe:10832): WARNING **: 13:06:01.849: Failed to load shared library 'libgdk-3-0.dll' referenced by the typelib: 'libgdk-3-0.dll': The specified procedure could not be found.
Traceback (most recent call last):
  File "tartube", line 32, in <module>
    from lib import mainapp
  File "C:\msys64\home\slart\tartube\lib\mainapp.py", line 26, in <module>
    from gi.repository import Gtk, GObject, GdkPixbuf
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "C:\msys64\mingw64\lib\python3.7\site-packages\gi\importer.py", line 145, in load_module
    importlib.import_module('gi.repository.' + dep.split("-")[0])
  File "C:\msys64\mingw64\lib\python3.7\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 668, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 638, in _load_backward_compatible
  File "C:\msys64\mingw64\lib\python3.7\site-packages\gi\importer.py", line 146, in load_module
    dynamic_module = load_overrides(introspection_module)
  File "C:\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\__init__.py", line 118, in load_overrides
    override_mod = importlib.import_module(override_package_name)
  File "C:\msys64\mingw64\lib\python3.7\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "C:\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\Gdk.py", line 83, in <module>
    Color = override(Color)
  File "C:\msys64\mingw64\lib\python3.7\site-packages\gi\overrides\__init__.py", line 195, in override
    assert g_type != TYPE_NONE
AssertionError

C:\msys64\home\slart\tartube>PAUSE
Press any key to continue . . .

**Comment by axcore** [comment]
Ok, that rules out a problem with the installer. Let's try running tartube from within the msys2 environement, but without opening the terminal window.

In the Tartube directory, create a bash script:

C:\msys2\home\YOURNAME\tartube\startme.sh

(Obviously I don't need to point out that MS Windows will create a file called startme.sh.txt if you're not using a proper text editor)

Open the script in your text editor, and add these three lines:

#!/bin/bash
cd tartube
python3 tartube

Save the script. Now open a command terminal, and type this:

c:\msys64\usr\bin\mintty.exe -h always /bin/env MSYSTEM=MINGW64 /bin/bash -lc /home/YOURNAME/tartube/startme.sh

**Comment by slartie** [comment]
No worries about the editor. I've been using emacs for ages :)

I created the bash script and ran it with the mintty command, and it executes just fine, so that works fine.

**Comment by axcore** [comment]
>No worries about the editor. I've been using emacs for ages :)
I have to explain things in easy steps, in case other people are reading and trying the same thing :)

Tomorrow I will create another windows installer that includes the bash shell, rather than a Windows .bat file.

**Comment by axcore** [comment]
I create a new installer, v0.7.0. 

I am still trying to work out, what to include and what to exclude, so I hope the installer works (but I can't guarantee it)

**Comment by palexGit** [comment]
hi,

I just tried the 0.7 installer for Win10 64bits, and it's don't work.
The installation works well, but when à click on "Run" in the end that's do nothing.
Same thing when i use the link on my desktop.

**Comment by slartie** [comment]
The startup command needs to be tweaked ... or the tartube folder will have to be moved

msys starts the user in c:\users\username .. that's the home folder, so...

..\..\..\usr\bin\mintty.exe -w hide /bin/env MSYSTEM=MINGW64 /bin/bash -lc /home/user/tartube/tartube_mswin.sh

...won't work.

**Comment by axcore** [comment]
palexGit, what is the error message you see, when you run Start Menu > Tartube > Test gtk graphics ?

**Comment by palexGit** [comment]
Axcore,

here's the result of gtk test : 

`/etc/post-install/01-devices.post: line 56: chmod: command not found
/etc/post-install/01-devices.post: line 85: chmod: command not found
/etc/post-install/01-devices.post: line 90: /usr/bin/test: No such file or directory
/etc/post-install/01-devices.post: line 90: ln: command not found
/etc/post-install/01-devices.post: line 91: /usr/bin/test: No such file or directory
/etc/post-install/01-devices.post: line 91: ln: command not found
/etc/post-install/01-devices.post: line 92: /usr/bin/test: No such file or directory
/etc/post-install/01-devices.post: line 92: ln: command not found
/etc/post-install/01-devices.post: line 93: /usr/bin/test: No such file or directory
/etc/post-install/01-devices.post: line 93: ln: command not found
/etc/post-install/03-mtab.post: line 5: ln: command not found
/etc/post-install/05-home-dir.post: line 4: exec: id: not found
/etc/post-install/05-home-dir.post: line 224: mkdir: command not found
/home/palex could not be created.
Setting HOME to /tmp.
/etc/profile: line 128: /usr/bin/hostname: No such file or directory
/etc/profile.d/lang.sh: line 19: /usr/bin/locale: No such file or directory
/etc/profile.d/tzset.sh: line 22: /usr/bin/tzset: No such file or directory
/home/user/tartube/hello_world_mswin.sh: line 4: cd: tartube: No such file or directory
C:\Users\palex\AppData\Local\Tartube\msys64\mingw64\bin\python3.exe: can't open file 'hello_world.py': [Errno 2] No such file or directory
/bin/env: Exit 2.
`

**Comment by axcore** [comment]
palexGit, 

It looks like this is is the source of your problems:

/etc/post-install/05-home-dir.post: line 224: mkdir: command not found
/home/palex could not be created.
Setting HOME to /tmp.

Can you please try the **TEST-1** installer, and tell me the error messages (if any) from both the Gtk graphics test, and Tartube itself.

https://sourceforge.net/projects/tartube/files/v0.7.0%20TESTING%20ONLY/install-tartube-0.7.0-64bit-test1.exe/download

**Comment by palexGit** [comment]
OK,
i've unisntalled le previous version and installed TEST-1.

Here is te the return of gtk Grafiks test and tartube itself. It's still doesn't work

Gtk Graphics test

Copying skeleton files.
These files are for the users to personalise their msys2 experience.

They will never be overwritten nor automatically updated.

/etc/post-install/05-home-dir.post: line 233: /usr/bin/find: No such file or directory
/home/user/tartube/hello_world_mswin.sh: line 4: cd: tartube: No such file or directory
C:\Users\palex\AppData\Local\Tartube\msys64\mingw64\bin\python3.exe: can't open file 'hello_world.py': [Errno 2] No such file
or directory


Tartube Itself
/home/user/tartube/tartube_mswin.sh: line 4: cd: tartube: No such file or directory
C:\Users\palex\AppData\Local\Tartube\msys64\mingw64\bin\python3.exe: can't open file 'tartube': [Errno 2] No such file or
directory

**Comment by axcore** [comment]
slartie,

>The startup command needs to be tweaked ... or the tartube folder will have to be moved
>msys starts the user in c:\users\username .. that's the home folder, so...

I'm not sure that's right. I tried this test, do you get different results?

1. I downloaded and  installed a fresh copy of msys2. It installs itself in

C:\msys64

2. I opened the mingw64 terminal by double-clicking C:\msys64\mingw64.exe
3. In the terminal, I typed 

mkdir wibble

4. This directory appears in C:\msys64\home\YOURNAME\wibble. It doesn't appear in C:\Users\YOURNAME

I also tried the Tartube installer, installing it to C:\Tartube (as you did). Nothing is installed in C:\Users\YOURNAME; the home directory used by msys2 appears to be C:\msys64\home\YOURNAME

If we need to tweak the startup command to use absolute paths, I can make the Tartube installer create the .BAT file itself. To use the absolute path, we only need to remove the colon, following the example in https://stackoverflow.com/questions/47438779/executing-a-script-in-msys2-mingw

In other words, the installer could write a .BAT file containing

......\usr\bin\mintty.exe -h always /bin/env MSYSTEM=MINGW64 /bin/bash -lc /c/Tartube/msys6/home/user/tartube/tartube_mswin.sh

**Comment by axcore** [comment]
palexGit,

I can't repoduce your errors, so I am working my eyes closed here :)

Try TEST2 and, if that doesn't work, TEST3

https://sourceforge.net/projects/tartube/files/v0.7.0%20TESTING%20ONLY/

**Comment by palexGit** [comment]
OK 
For the installer Test 2 i have : 
gtk
Copying skeleton files.
These files are for the users to personalise their msys2 experience.

They will never be overwritten nor automatically updated.

/etc/post-install/05-home-dir.post: line 236: /usr/bin/install: No such file or directory
/etc/post-install/05-home-dir.post: line 236: /usr/bin/install: No such file or directory
/etc/post-install/05-home-dir.post: line 236: /usr/bin/install: No such file or directory
/etc/post-install/05-home-dir.post: line 236: /usr/bin/install: No such file or directory
/etc/post-install/05-home-dir.post: line 236: /usr/bin/install: No such file or directory
/home/user/tartube/hello_world_mswin.sh: line 4: cd: tartube: No such file or directory
C:\Users\palex\AppData\Local\Tartube\msys64\mingw64\bin\python3.exe: can't open file 'hello_world.py': [Errno 2] No such file
or directory

Tartube
/home/user/tartube/tartube_mswin.sh: line 4: cd: tartube: No such file or directory
C:\Users\palex\AppData\Local\Tartube\msys64\mingw64\bin\python3.exe: can't open file 'tartube': [Errno 2] No such file or
directory

For Installer TEST 3
Gtk Test
Copying skeleton files.
These files are for the users to personalise their msys2 experience.

They will never be overwritten nor automatically updated.

'./.bashrc' -> '/home/palex/.bashrc'
'./.bash_logout' -> '/home/palex/.bash_logout'
'./.bash_profile' -> '/home/palex/.bash_profile'
'./.inputrc' -> '/home/palex/.inputrc'
'./.profile' -> '/home/palex/.profile'
/home/user/tartube/hello_world_mswin.sh: line 4: cd: tartube: No such file or directory
C:\Users\palex\AppData\Local\Tartube\msys64\mingw64\bin\python3.exe: can't open file 'hello_world.py': [Errno 2] No such file
or directory
/bin/env: Exit 2.

Tartube
/home/user/tartube/tartube_mswin.sh: line 4: cd: tartube: No such file or directory
C:\Users\palex\AppData\Local\Tartube\msys64\mingw64\bin\python3.exe: can't open file 'tartube': [Errno 2] No such file or
directory

Test 3 is better but there's a problem with directories creation.

Maybe i can see the script and test on my PC without installing anything ?

**Comment by axcore** [comment]
The installation script for MS Windows is in the /nsis folder, i.e.

https://github.com/axcore/tartube/blob/master/nsis/tartube_install_64bit.nsi

The script describes the whole procedure, but it's out of date. This is what I am doing now:

1. Install msy2 (64 bit version, download from msys2.org)
2. Run the minw64 terminal, C:\msys64\mingw64.exe
3. In the terminal, type 
pacman -Syu
4. If the terminal wants to close, close it
5. Open the terminal again, and type these commands
pacman -Su
pacman -S mingw-w64-x86_64-python3 
pacman -S mingw-w64-x86_64-python3-pip
pacman -S mingw-w64-x86_64-python3-gobject
pacman -S mingw-w64-x86_64-python3-requests
pacman -S mingw-w64-x86_64-gtk3 
pacman -S mingw-w64-x86_64-gsettings-desktop-schemas
6. Copy the tartube source code into C:\msys64\home\YOURNAME (so, you will have a folder called C:\msys64\home\YOURNAME\tartube and a script called C:\msys64\home\YOURNAME\tartube\hello_world.py)
7. In the minwg64 terminal type 'python3 tartube'

From there, you can create your own MS Windows installer, if you really want to

8. Create a new folder, C:\dist (the name doesn't matter)
9. Copy the whole C:\msys64 folder into C:\dist, so you now have a C:\dist\msys64 folder
10. Go to C:\dist\msys64\home\YOURNAME\tartube\nsis. There are four windows .BAT files there. Move them into the folder above
11. Then COPY the other files in that folder, from C:\dist\msys64\home\YOURNAME\tartube\nsis to C:\dist
12. Now install NSIS, from nsis.sourceforge.io
13. Go to C:\dist, right click the installer script, and select 'Compile installer' (I can't remember what the exact text is)
14. After a few minutes, the installer appears in C:\dist

**Comment by palexGit** [comment]
Thanks,

Before using  nsis, i think i'll try to create tartube.exe with pyinstaller for example.

if all work well, i'll look for nsis.

**Comment by axcore** [comment]
>Before using nsis, i think i'll try to create tartube.exe with pyinstaller for example.
You can try, but pyinstaller is known not to work with PyGObject (unless it has been updated recently)

**Comment by slartie** [comment]
When I get back to my system, I'll try clearing everything out and starting over and report back.

**Comment by axcore** [comment]
Fixed in v1.2.0
